src/compiler/utilities.ts

```diff
@@ -1529,22 +1596,22 @@ namespace ts {
         else if (lhs.expression.kind === SyntaxKind.ThisKeyword) {
             return SpecialPropertyAssignmentKind.ThisProperty;
         }
-        else if (lhs.expression.kind === SyntaxKind.PropertyAccessExpression) {
+        else if (isPropertyAccessExpression(lhs.expression)) {
             // chained dot, e.g. x.y.z = expr; this var is the 'x.y' part
-            const innerPropertyAccess = <PropertyAccessExpression>lhs.expression;
-            if (innerPropertyAccess.expression.kind === SyntaxKind.Identifier) {
+            if (isIdentifier(lhs.expression.expression)) {
                 // module.exports.name = expr
-                const innerPropertyAccessIdentifier = <Identifier>innerPropertyAccess.expression;
-                if (innerPropertyAccessIdentifier.escapedText === "module" && innerPropertyAccess.name.escapedText === "exports") {
+                if (lhs.expression.expression.escapedText === "module" && lhs.expression.name.escapedText === "exports") {
                     return SpecialPropertyAssignmentKind.ExportsProperty;
                 }
-                if (innerPropertyAccess.name.escapedText === "prototype") {
+                if (lhs.expression.name.escapedText === "prototype") {
                     return SpecialPropertyAssignmentKind.PrototypeProperty;
                 }
             }

```

Comment:
The preceding changes in this function are just a refactor, *except* for the following 3 lines.